### PR TITLE
[kind] fix kind deployment

### PR DESCRIPTION
**Pull request by pducolin**
What does this PR do?
---------------------

* fix kind install command
* use AmazonECS family OS by default for Kind
* adds inv create-kind task
* adds integration test for kind

Which scenarios this will impact?
-------------------

kind

Motivation
----------

I broke kind install command with https://github.com/DataDog/test-infra-definitions/pull/671, this PR fixes it and adds an integration test for kind vm

Additional Notes
----------------

I tried to run tests in parallel with `t.Parallel` in subtests, but it failed, might race on config file. To be investigated.

### CODE REVIEW

File: components/kubernetes/kind.go [hunk 1]
```diff
@@ -38,7 +38,12 @@ func NewKindCluster(env config.CommonEnvironment, vm *remote.Host, resourceName,
 			return err
 		}
 
-		_, dockerInstallCmd, err := docker.NewManager(env, vm, true, opts...)
+		shouldInstallDocker := true
+		// docker is installed by default on Amazon Linux ECS
+		if vm.OS.Descriptor().Flavor == os.AmazonLinuxECS {
+			shouldInstallDocker = false
+		}
+		_, dockerInstallCmd, err := docker.NewManager(env, vm, shouldInstallDocker, opts...)
```
Comment (vboulineau) [inline components/kubernetes/kind.go:46]: We need to add a `TODO: Lazy Docker install`

**Review by vboulineau — APPROVED** [review]

File: components/kubernetes/kind.go (hunk 1 above)
Comment (pducolin) [reply components/kubernetes/kind.go:46]: We have a card covering it

Comment (pducolin) [reply components/kubernetes/kind.go:46]: https://datadoghq.atlassian.net/browse/ADXT-88, part of this sprint

**Comment by pducolin** [comment]
I fixed the parallel running

**Comment by pducolin** [comment]
Getting limited by memory, will increase limits

**Comment by pducolin** [comment]
Giving up to parallel test execution, it takes over 16GiB to run all 3 tests, might come back later and run docker + vm in // and kind in series

**Pull request merged**